### PR TITLE
Fixes #27925 - Change Models table pagination to react

### PR DESCRIPTION
**Pull request by MariaAga**
Uses https://github.com/theforeman/foreman/pull/7061 pagination in react instead from the erb

### CODE REVIEW

**Comment by xprazak2** [comment]
Seems to work as expected but request action gets triggered twice on page change and initial load as well:

models-reqs

**Comment by MariaAga** [comment]
@xprazak2  Thanks for noticing, fixed in the pagination PR

File: webpack/assets/javascripts/react_app/components/ModelsTable/ModelsTable.js [hunk 1]
```diff
@@ -15,10 +15,13 @@ const ModelsTable = ({
...
   status,
   results,
+  itemCount,
+  pagination,
+  onPaginationChange,
 }) => {
   useEffect(() => {
     getTableItems(getURIQuery(window.location.href));
-  }, []);
+  }, [pagination.page, pagination.perPage]);
```
Comment (xprazak2) [inline webpack/assets/javascripts/react_app/components/ModelsTable/ModelsTable.js:24]: I found out re-running the hook when `perPage` changes this still causes the the double calls under certain conditions:

1) set 'Entries per page' under Settings -> General tab to a non-default value, 50 will work
2) refresh the models page

The initial value is 20, but it gets updated after the API call finishes, which triggers another call.

**Comment by xprazak2** [comment]
@MariaAga, thanks for a quick update, I left one comment inline.

**Comment by MariaAga** [comment]
as the default per page comes from the server right now, after the api call the per page will change (to the default value) and will trigger a reload to the data.
So for now I'm closing this PR

**Pull request closed without merging**